src/event-storage.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+/* global window */
+
+const storageKey = 'fusion-events';
+
+type EventType = {|
+  type: mixed,
+  payload: mixed,
+|};
+
+// localStorage wrappers *exported only for testing*
+export const get = () => {
+  try {
+    const events = JSON.parse(window.localStorage.getItem(storageKey));
+    return Array.isArray(events) ? events : [];
+  } catch (e) {
+    return [];
+  }
+};
+
+export const clear = () => {
+  try {
+    window.localStorage.removeItem(storageKey);
+  } catch (e) {
+    // storage may not be writable, do nothing
+  }
+};
+
+export const set = (events: EventType[]) => {
+  try {
+    window.localStorage.setItem(storageKey, JSON.stringify(events));
+  } catch (e) {
+    // storage might be full or not writable, do nothing
```

Comment:
seems like we lose the events if we somehow cannot write the store and prevent them for being sent... Not sure if that's robust?

Reply:
Thats a good point, I'll look into this.